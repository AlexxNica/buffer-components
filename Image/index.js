import React, { PropTypes } from 'react';

const Image = ({ src, alt, width, height }) =>
  <img
    src={src}
    alt={alt}
    width={width}
    height={height}
  />;

Image.propTypes = {
  alt: PropTypes.string,
  height: PropTypes.string,
  src: PropTypes.string.isRequired,
  width: PropTypes.string,
};

Image.defaultProps = {
  alt: 'Image',
};

export default Image;
