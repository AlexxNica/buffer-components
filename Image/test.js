jest.unmock('./index');
import React from 'react';
import { renderAndCheckSnapshot } from '../testHelpers';
import Image from './index';

describe('Image', () => {
  it('Image component', () => {
    renderAndCheckSnapshot(<Image src={"/cool/image.png"} />);
  });

  it('Image w/ custom alt text', () => {
    renderAndCheckSnapshot(
      <Image
        src={"/cool/image.png"}
        alt={"Alternate"}
      />
    );
  });

  it('Image w/ width', () => {
    renderAndCheckSnapshot(
      <Image
        src={"/cool/image.png"}
        width={"50px"}
      />
    );
  });

  it('Image w/ height', () => {
    renderAndCheckSnapshot(
      <Image
        src={"/cool/image.png"}
        height={"50px"}
      />
    );
  });
});
