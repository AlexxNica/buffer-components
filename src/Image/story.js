import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Image from './index';

storiesOf('Image')
  .add('Default', () => (
    <Image src={'https://buffer.com/images/app/buffer-logo@2x.png'} />
  ))
  .add('W=100%', () => (
    <Image
      src={'https://buffer.com/images/app/buffer-logo@2x.png'}
      width={"100%"}
    />
  ))
  .add('W=200px H=40px', () => (
    <Image
      src={'https://buffer.com/images/app/buffer-logo@2x.png'}
      width={"200px"}
      height={"40px"}
    />
  ));
