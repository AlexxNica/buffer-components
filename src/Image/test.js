jest.unmock('./index');
import React from 'react';
import Image from './index';
import renderer from 'react-test-renderer';

describe('Image', () => {
  it('Image component', () => {
    const component = renderer.create(
      <Image src={"/cool/image.png"} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Image w/ custom alt text', () => {
    const component = renderer.create(
      <Image
        src={"/cool/image.png"}
        alt={"Alternate"}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Image w/ width', () => {
    const component = renderer.create(
      <Image
        src={"/cool/image.png"}
        width={"50px"}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Image w/ height', () => {
    const component = renderer.create(
      <Image
        src={"/cool/image.png"}
        height={"50px"}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
