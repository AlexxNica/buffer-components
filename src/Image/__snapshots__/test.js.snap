exports[`Image Image component 1`] = `
<img
  alt="Image"
  height={undefined}
  src="/cool/image.png"
  width={undefined} />
`;

exports[`Image Image w/ custom alt text 1`] = `
<img
  alt="Alternate"
  height={undefined}
  src="/cool/image.png"
  width={undefined} />
`;

exports[`Image Image w/ height 1`] = `
<img
  alt="Image"
  height="50px"
  src="/cool/image.png"
  width={undefined} />
`;

exports[`Image Image w/ width 1`] = `
<img
  alt="Image"
  height={undefined}
  src="/cool/image.png"
  width="50px" />
`;
